### PR TITLE
Extend cl.array.(maximum|minimum|if_positive) to take scalars

### PR DESCRIPTION
**Pull request by kaushikcfd**

### CODE REVIEW

File: pyopencl/array.py [hunk 1]
```diff
@@ -42,6 +42,9 @@
         get_common_dtype as _get_common_dtype_base)
 from pyopencl.characterize import has_double_support
 from pyopencl import cltypes
+from numbers import Number
+
+SCALAR_CLASSES = (Number, np.number, np.bool_, bool)
```
Comment (inducer) [inline pyopencl/array.py]: Doesn't `isinstance(x, np.number)` imply `isinstance(x, Number)`?

Comment (inducer) [inline pyopencl/array.py]: Add a test?

File: pyopencl/array.py [hunk 2]
```diff
@@ -2673,6 +2676,17 @@ def if_positive(criterion, then_, else_, out=None, queue=None):
...
     """
 
+    if all(isinstance(k, SCALAR_CLASSES) for k in [criterion,
+                                                   then_,
+                                                   else_]):
+        result = np.where(criterion, then_, else_)
+
+        if out is not None:
+            out[...] = result
+            return
```
Comment (inducer) [inline pyopencl/array.py]: The non-scalar code path unconditionally returns `out`.

File: pyopencl/array.py [hunk 3]
```diff
@@ -2673,6 +2676,17 @@ def if_positive(criterion, then_, else_, out=None, queue=None):
     contains *then_[i]* if *criterion[i]>0*, else *else_[i]*.
     """
 
+    if all(isinstance(k, SCALAR_CLASSES) for k in [criterion,
+                                                   then_,
+                                                   else_]):
```
Comment (inducer) [inline pyopencl/array.py]: This'll probably be quicker to execute as `isinstance() and isinstance() and isinstance()` (no separate stack frame setup, for instance).

**Review by inducer — COMMENTED** [review]
Thanks! A few nits to pick below.

File: pyopencl/array.py (hunk 1 above)
Comment (kaushikcfd) [reply pyopencl/array.py]: Yep removed the redundancy here. Thanks! 21fa507

Comment (kaushikcfd) [reply pyopencl/array.py]: Yep, thanks! 06c1e0d

File: pyopencl/array.py (hunk 2 above)
Comment (kaushikcfd) [reply pyopencl/array.py]: Whoops fixed in <del>f706943</del>991135bfddf304b082b790912a7e76c65f48d176.

File: pyopencl/array.py (hunk 3 above)
Comment (kaushikcfd) [reply pyopencl/array.py]: <del>9e9571f</del> eb29e2911c7c846fd9e1610df80d49459d91ea01

**Comment by kaushikcfd** [comment]
@inducer: Thanks a lot for the suggestions. I believe this is ready.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**